### DISCUSSION TITLE
Is there a particular reason that nested SUB_WORKFLOW within a DO_WHILE is not supported?

### DISCUSSION THREAD

**Discussion by pweatherby-ccctc** [Netflix/conductor#2532] [Q&A] [answered]
The [documentation](https://netflix.github.io/conductor/configuration/systask/#do-while-task) on DO_WHILE tasks clearly states that SUB_WORKFLOW within a DO_WHILE is not supported.

The [WorkflowTaskTypeConstraint](https://github.com/Netflix/conductor/blob/v2.31.4/core/src/main/java/com/netflix/conductor/validations/WorkflowTaskTypeConstraint.java#L152) enforces that DO_WHILE loopOver tasks are not (and do not contain) a SUB_WORKFLOW.

My question is: "Why is not supported?" 

Perhaps this has already been answered/documented, but I have searched and cant find reasoning on it.

I am using v2.31.4, but plan to upgrade soon-ish. but it appears that, even at the latest, it's still not supported.

Perhaps it has to do with task domains? or isolation concerns? or... i dunno? 
Or perhaps it is a hold-over from a limitation of a previous workflow engine that no longer exists? (and now, we can support it?)

EDIT/Update: I was still searching more for an answer to this question and found: https://github.com/Netflix/conductor/pull/1435/files which was an attempt (that was eventually closed due to inactivity) to add in support. Whats interesting about it (to me at least) is: the code changed is all removing superficial checks/tests that impose the unsupported constraint. I don't see any code that adds functionality per-say. Instead, just removing the code that dis-allows it. Reading the comments on the PR, it looks like everything was good to go, except needing some additional integration tests.

@apanicker-nflx @manan164 Can you confirm any of the above regarding the status of nested sub_workflows within a do_while?

**Comment by manan164** [comment, marked as answer]
Hi @pweatherby-ccctc , I had created [PR](https://github.com/Netflix/conductor/pull/1435) to support it. Had to close it since I was not able to find time to write integration tests. Feel free to write and use it.
